### PR TITLE
chore/9430-RNUpdate

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Updated RN from 0.73.8 to 0.75.2

Also updated RN Gesture Handler, Safe area context, screens, and react along with the configs in dev dependencies

Updated gemfile, pods, and android gradle, and android studio to Koala

Small box component change that should have no affect

Things to note

with 0.74 state updates are now batched together to trigger less re renders. Could potentially be breaking but I don't think we had any state variables that updated another state variable as a result of it changing. 

detox runs

https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/10636519873

https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/10636522285

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
RN is updated to 0.75.2 and the app still works on iOS and Android and Detox/Unit tests still runs

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by alexandec — DISMISSED** [review]
Approved with a couple questions

File: VAMobile/ios/Podfile [hunk 1]
```diff
@@ -37,7 +37,8 @@ target 'VAMobile' do
     react_native_post_install(
       installer,
       config[:reactNativePath],
-      :mac_catalyst_enabled => false
+      :mac_catalyst_enabled => false,
+      # :ccache_enabled => true
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/ios/Podfile:41]: yeah, it was in the update helper, so I left it in

File: VAMobile/e2e/tests/Messages.e2e.ts [hunk 2]
```diff
@@ -355,7 +355,7 @@ describe('Messages Screen', () => {
     await element(by.text(CommonE2eIdConstants.VETERAN_CRISIS_LINE_BTN_TEXT)).tap()
     await expect(element(by.text('Veterans Crisis Line'))).toExist()
     await element(by.text('Done')).tap()
-  })
+  // })
```
Comment (alexandec) [inline VAMobile/e2e/tests/Messages.e2e.ts:358]: Is this comment intentional?

**Review by alexandec — DISMISSED** [review]
Approving with a question

File: VAMobile/e2e/tests/Messages.e2e.ts (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/e2e/tests/Messages.e2e.ts:358]: good catch

I think this may have been when Rachael was trying to fix them

**Review by alexandec — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Pull request merged**